### PR TITLE
age-plugin-ledger: init at 0.1.2

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Description of changes

This is a plugin for `age` and `rage` to use a Ledger Nano as a hardware security enclave.
It is highly similar to `age-plugin-yubikey` which is already present in Nixpkgs.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 231275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>age-plugin-ledger</li>
  </ul>
</details>

**Review by Janik-Haag — COMMENTED** [review]
Is it supposed to panick here?

```
age-plugin-ledger -l
thread 'main' panicked at 'no Ledger Nano were found', src/hid.rs:15:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```
age-plugin-ledger -i
thread 'main' panicked at 'no Ledger Nano were found', src/hid.rs:15:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

EDIT: nevermind, just me being stupid, but I can't test it since I don't have the hardware needed for that. You can post the pr in the ready for review discourse post.

**Comment by Janik-Haag** [comment]
@erdnaxe edited my comment. 
Can you do a rebase?

**Comment by Janik-Haag** [comment]
@erdnaxe are you still interested?

**Comment by erdnaxe** [comment]
> @erdnaxe are you still interested?

Yes, but I haven't found the time yet to retest the whole setup. I believe that now the Ledger Nano app for age is directly available in Ledger Live which should greatly simplify the setup.

**Comment by erdnaxe** [comment]
Updated to 0.1.2. I tested it using Ledger Nano X and S+, both are working great.

File: pkgs/tools/security/age-plugin-ledger/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    libusb1
+    openssl
+  ] ++ lib.optional stdenv.isDarwin AppKit;
+
+  nativeCheckInputs = [
+    rage
+  ];
+
+  meta = with lib; {
+    description = "Ledger Nano plugin for age";
```
Comment (Janik-Haag) [inline pkgs/tools/security/age-plugin-ledger/default.nix:39]:
```suggestion
    description = "A Ledger Nano plugin for age";
```

File: pkgs/tools/security/age-plugin-ledger/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+, AppKit
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "age-plugin-ledger";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "Ledger-Donjon";
+    repo = pname;
```
Comment (Janik-Haag) [inline pkgs/tools/security/age-plugin-ledger/default.nix:18]:
```suggestion
    repo = "age-plugin-ledger";
```

we recently started using the repo name as string even if it's the same.

**Review by Janik-Haag — CHANGES_REQUESTED** [review]
just some small nitpicks

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**